src/main_test.go

```diff
@@ -0,0 +1,396 @@
+				"allowed":false
+			}
+			}`))
+}
+
+func TestWriteUnprotectedResourcesAllowed(t *testing.T) {
+	accessTest(t, defaultAuthorizer, false,
+		[]byte(
+			`{
+			"kind":"SubjectAccessReview",
+			"apiVersion":"authorization.k8s.io/v1",
+			"spec":{
+				"resourceAttributes":{
+					"namespace":"safe-namespace",
+					"verb":"delete",
+					"version":"v1",
+					"resource":"pods",
+					"name":"generic-pod"
+				},
+				"user":"kubernetes-not-admin",
+				"groups":["system:authenticated"]
+			},
+			"status":{
+				"allowed":false
+			}
+			}`))
+}
+
+func TestEmptyNamespacesRequestsDenied(t *testing.T) {
+	accessTest(t, defaultAuthorizer, true,
+		[]byte(
+			`{
+			"kind":"SubjectAccessReview",
+			"apiVersion":"authorization.k8s.io/v1",
+			"spec":{
+				"resourceAttributes":{
+					"namespace":"",
+					"verb":"get",
+					"version":"v1",
+					"resource":"secrets",
+					"name":"important-creds"
+				},
+				"user":"not-admin",
+				"groups":["group1"]
+			},
+			"status":{
+				"allowed":false
+			}
+			}`))
+}
+
+func TestProtectedReadAllRequestsDenied(t *testing.T) {
+	accessTest(t, defaultAuthorizer, true,
+		[]byte(
+			`{
+			"kind":"SubjectAccessReview",
+			"apiVersion":"authorization.k8s.io/v1",
+			"spec":{
+				"resourceAttributes":{
+					"namespace":"kube-system",
+					"verb":"get",
+					"version":"v1",
+					"resource":"*",
+					"name":"important-creds"
+				},
+				"user":"not-admin",
+				"groups":["group1"]
+			},
+			"status":{
+				"allowed":false
+			}
+			}`))
+}
+
+func TestProtectedWriteAllRequestsDenied(t *testing.T) {
+	accessTest(t, defaultAuthorizer, true,
+		[]byte(
+			`{
+			"kind":"SubjectAccessReview",
+			"apiVersion":"authorization.k8s.io/v1",
+			"spec":{
+				"resourceAttributes":{
+					"namespace":"kube-system",
+					"verb":"delete",
+					"version":"v1",
+					"resource":"*",
+					"name":"important-creds"
+				},
+				"user":"not-admin",
+				"groups":["group1"]
+			},
+			"status":{
+				"allowed":false
+			}
+			}`))
+}
+
+func TestProtectedAllVerbRequestsDenied(t *testing.T) {
+	accessTest(t, defaultAuthorizer, true,
+		[]byte(
+			`{
+			"kind":"SubjectAccessReview",
+			"apiVersion":"authorization.k8s.io/v1",
+			"spec":{
+				"resourceAttributes":{
+					"namespace":"kube-system",
+					"verb":"*",
+					"version":"v1",
+					"resource":"secrets",
+					"name":"important-creds"
+				},
+				"user":"not-admin",
+				"groups":["group1"]
+			},
+			"status":{
+				"allowed":false
+			}
+			}`))
+}
+
+func TestProtectedAllVerbNonSecretRequestsDenied(t *testing.T) {
+	accessTest(t, defaultAuthorizer, true,
+		[]byte(
+			`{
+			"kind":"SubjectAccessReview",
+			"apiVersion":"authorization.k8s.io/v1",
+			"spec":{
+				"resourceAttributes":{
+					"namespace":"kube-system",
+					"verb":"*",
+					"version":"v1",
+					"resource":"pods",
+					"name":"important-creds"
+				},
+				"user":"not-admin",
+				"groups":["group1"]
+			},
+			"status":{
+				"allowed":false
+			}
+			}`))
+}
```

Comment:
I think we need to test sending invalid inputs, and see what we get, to make sure its expected (it should be 400 bad request, I think)